Task3/Task3_Jordan_Petrov/src/python/car_wash/dbpool.py

```diff
@@ -0,0 +1,159 @@
+        self._pool_size = pool_size
+        self._lock = Lock()
+
+    def _create_connection(self) -> psycopg2.connect:
+        """Create connection to DB"""
+        return psycopg2.connect(dbname=self._dbname,
+                                user=self._username,
+                                password=self._password,
+                                host=self._hostname)
+
+    def _get_connection(self) -> psycopg2.connect:
+        """Get connection from pool"""
+
+        while True:
+            with self._lock:
+                if len(self._pool) > 0:
+                    return self._pool.pop()
+                else:
+                    time.sleep(1)
+
+    def _put_connection(self, conn: psycopg2.connect) -> None:
+        """Put connection back to pool"""
+        with self._lock:
+            self._pool.append(conn)
+
+    @contextmanager
+    def connection(self):
+        """
+        Context manager for the read only pool.
+        Use cursor to execute queries.
+        """
+        if len(self._pool) < self._pool_size:
+            self._pool.append(self._create_connection())
+
+        conn = self._get_connection()
+        cursor_ = conn.cursor()
+        try:
+            yield cursor_
+        except psycopg2.Error:
+            logging.error('Invalid query.')
+            raise psycopg2.Error('Invalid query.')
+        finally:
+            self._put_connection(conn)
+
+
+class RWDBPool(DBPoolBase):
+    """Read only pool of connections."""
+
+    def __init__(self, hostname: str, password: str, username: str, _dbname: str, pool_size: int) -> None:
+        """Constructor of RODBPool connections
+
+        :Parameters:
+            - `hostname` - DB hostname
+        """
+        super().__init__()
+        self._pool = []
+        self._hostname = hostname
+        self._password = password
+        self._username = username
+        self._dbname = _dbname
+        self._pool_size = pool_size
+        self._lock = Lock()
+
+    def _create_connection(self) -> psycopg2.connect:
+        """Create connection to DB"""
+        return psycopg2.connect(dbname=self._dbname,
+                                user=self._username,
+                                password=self._password,
+                                host=self._hostname)
+
+    def _get_connection(self) -> psycopg2.connect:
+        """Get connection from pool"""
+
+        while True:
+            with self._lock:
+                if len(self._pool) > 0:
+                    return self._pool.pop()
+                else:
+                    time.sleep(1)
+
+    def _put_connection(self, conn: psycopg2.connect) -> None:
+        """Put connection back to pool"""
+        with self._lock:
+            self._pool.append(conn)
+
+    @contextmanager
+    def transaction(self):
+        """
+        Context manager for the read write pool.
+        Use cursor to execute queries.
+        """
+
+        if len(self._pool) < self._pool_size:
```

Comment:
General comment: you do not know how many connections were created. You need to have counter.